plugins/package-managers/node/src/main/kotlin/NodePackageManager.kt

```diff
@@ -77,7 +78,15 @@ abstract class NodePackageManager(val managerType: NodePackageManagerType) : Pac
         analysisRoot: File,
         definitionFiles: List<File>,
         analyzerConfig: AnalyzerConfiguration
-    ) = NodePackageManagerDetection(definitionFiles).filterApplicable(managerType)
+    ): List<File> {
+        val enabledIds = analyzerConfig.determineEnabledPackageManagers().map { it.descriptor.id.uppercase() }
```

Comment:
commit: Why do you believe it's better to map the definition files only to _enabled managers_, vs. mapping the definition files to _all node managers_ (but skipping the analysis of the disabled ones)?

Reply:
> I'm unsure whether you're misinterpreting

I've indeed mis-interpreted the intention a bit. But my comment IMHO still applies IIUC to the general case, not specific to:

> If there is only a package.json file with no lockfile, then there is no safe way for ORT to tell which package manager should be used for that file. Before this PR, always NPM was assigned to that file, even if NPM as a manager was disabled, which effectively skipped that file.

Reply:
So commenting on your

> I believe using a "different" package managers for the analysis can cause problems.

That's true, but given that the user explicitly opts-in to this behavior by only enabling specific Node package managers instead of all of them, I believe the solution is fine.

Reply:
Hm, maybe a different approach of discussion.
Would you mind outlining  the behavior before and after this PR for the following scenario?

-definition files in repository
  - PNPM project, manager can be unambiguously deteted
  - NPM project, manager can be unambiguously deteted
- Enabled managers: Npm (only)

Reply:
I.e. this PR changes nothing WRT unambiguous cases.

Reply:
Thanks!